shlr/gdb/src/gdbclient/core.c

```diff
@@ -82,20 +84,52 @@ static void reg_cache_init(libgdbr_t *g) {
 	}
 }
 
+static bool _isbreaked = false;
+
+static void gdbr_break_process(libgdbr_t *g) {
+	_isbreaked = true;
+}
+
+bool gdbr_lock_enter(libgdbr_t *g) {
+	r_cons_break_push (gdbr_break_process, g);
+	void *bed = r_cons_sleep_begin ();
+	r_th_lock_enter (g->gdbr_lock);
+	r_cons_sleep_end (bed);
+	if (_isbreaked) {
+		return false;
+	}
+	return true;
+}
+
+void gdbr_lock_leave(libgdbr_t *g) {
+	r_cons_break_pop ();
```

Comment:
I dont think u can call rcons from here

Reply:
It does work? Try removing it and run `dc`, you'll get stuck in an infinite loop.
Not sure why this would be an invalid usage, the implementation calls [r_cons_context_break_pop](https://github.com/radareorg/radare2/blob/master/libr/cons/cons.c#L333) with I.context which should be the same in enter and leave, no?